packages/custom_lint/test/create_project.dart

```diff
@@ -279,14 +293,21 @@ Directory createDartProject({
       if (analysisOptions != null) 'analysis_options.yaml': analysisOptions,
       if (pubspec != null) 'pubspec.yaml': pubspec,
       if (packageConfig != null)
-        '.dart_tool/package_config.json': packageConfig,
+        join('.dart_tool', 'package_config.json'): packageConfig,
     },
     name,
+    parent,
   );
 }
 
-Directory createTmpFolder(Map<String, String> files, String name) {
-  final newFolder = Directory.systemTemp.createTempSync(name);
+Directory createTmpFolder(
+    Map<String, String> files, String name, Directory? parent) {
+  late Directory newFolder;
+  if (parent == null) {
+    newFolder = Directory.systemTemp.createTempSync(name);
+  } else {
+    newFolder = Directory(join(parent.path, name))..createSync();
```

Comment:
I added the option to pass a parent directory all the way down here, otherwise there is no easy way to create multiple context roots, wich are required for the two tests that lead to the hanging process.